Sources/SpeziOnboarding/ConsentView/ConsentDocument+Export.swift

```diff
@@ -102,41 +62,193 @@ extension ConsentDocument {
         let markdown = await asyncMarkdown()
-        
+
         let markdownString = (try? AttributedString(
             markdown: markdown,
             options: .init(interpretedSyntax: .inlineOnlyPreservingWhitespace)
         )) ?? AttributedString(String(localized: "MARKDOWN_LOADING_ERROR", bundle: .module))
-        
-        let renderer = ImageRenderer(content: exportBody(markdown: markdownString))
-        let paperSize = CGSize(
+
+        let pageSize = CGSize(
             width: exportConfiguration.paperSize.dimensions.width,
             height: exportConfiguration.paperSize.dimensions.height
         )
-        renderer.proposedSize = .init(paperSize)
-        
+
+        let pages = paginatedViews(markdown: markdownString)
+
+        print("NumPages: \(pages.count)")
         return await withCheckedContinuation { continuation in
-            renderer.render { _, context in
-                var box = CGRect(origin: .zero, size: paperSize)
-                
-                /// Create in-memory `CGContext` that stores the PDF
-                guard let mutableData = CFDataCreateMutable(kCFAllocatorDefault, 0),
-                      let consumer = CGDataConsumer(data: mutableData),
-                      let pdf = CGContext(consumer: consumer, mediaBox: &box, nil) else {
-                    continuation.resume(returning: nil)
-                    return
-                }
-                
+            guard let mutableData = CFDataCreateMutable(kCFAllocatorDefault, 0),
+                  let consumer = CGDataConsumer(data: mutableData),
+                  let pdf = CGContext(consumer: consumer, mediaBox: nil, nil) else {
+                continuation.resume(returning: nil)
+                return
+            }
+
+            for page in pages {
                 pdf.beginPDFPage(nil)
-                pdf.translateBy(x: 0, y: 0)
-                
-                context(pdf)
+               
+                let hostingController = UIHostingController(rootView: page)
+                 hostingController.view.frame = CGRect(origin: .zero, size: pageSize)
+
+                 let renderer = UIGraphicsImageRenderer(bounds: hostingController.view.bounds)
+                 let image = renderer.image { ctx in
+                     hostingController.view.drawHierarchy(in: hostingController.view.bounds, afterScreenUpdates: true)
+                 }
+
+                pdf.saveGState()
+
+                pdf.translateBy(x: 0, y: pageSize.height)
+                pdf.scaleBy(x: 1.0, y: -1.0)
+
+                hostingController.view.layer.render(in: pdf)
+
+                pdf.restoreGState()
+                 
                 
                 pdf.endPDFPage()
-                pdf.closePDF()
-                
-                continuation.resume(returning: PDFDocument(data: mutableData as Data))
             }
+
+            pdf.closePDF()
+            continuation.resume(returning: PDFDocument(data: mutableData as Data))
```

Comment:
What is the continuation used for if you always resume it instantly?

Reply:
That was a remnant of the old code. I now removed it, thanks for pointing it out!